tests/shared/utils.spec.js

```diff
@@ -124,4 +127,191 @@ describe('Shared: Utils', function() {
+        path.basename = this.basename;
+        os.platform = this.platform;
+      });
+
+      it('should replace absolute paths with the basename', function() {
+        const object = {
+          prop1: 'some value',
+          prop2: '/home/john/rpi.img'
+        };
+
+        m.chai.expect(utils.hideObjectAbsolutePaths(object)).to.deep.equal({
+          prop1: 'some value',
+          prop2: 'rpi.img'
+        });
+
+      });
+
+      it('should replace nested absolute paths with the basename', function() {
+        const object = {
+          nested: {
+            path: '/home/john/rpi.img'
+          }
+        };
+
+        m.chai.expect(utils.hideObjectAbsolutePaths(object)).to.deep.equal({
+          nested: {
+            path: 'rpi.img'
+          }
+        });
+      });
+
+      it('should not alter /dev/sdb', function() {
+        const object = {
+          nested: {
+            path: '/dev/sdb'
+          }
+        };
+
+        m.chai.expect(utils.hideObjectAbsolutePaths(object)).to.deep.equal({
+          nested: {
+            path: '/dev/sdb'
+          }
+        });
+      });
+
+      it('should handle arrays', function() {
+        m.chai.expect(utils.hideObjectAbsolutePaths({
+          foo: 'foo',
+          bar: [ {
+            path: '/foo/bar/baz'
+          }, {
+            path: '/foo/bar/baz'
+          }, {
+            path: '/foo/bar/baz'
+          } ]
+        })).to.deep.equal({
+          foo: 'foo',
+          bar: [ {
+            path: 'baz'
+          }, {
+            path: 'baz'
+          }, {
+            path: 'baz'
+          } ]
+        });
+      });
+
+    });
+
+    describe('given Windows paths', function() {
+
+      beforeEach(function() {
+        this.isAbsolute = path.isAbsolute;
+        this.basename = path.basename;
+        this.platform = os.platform;
+        path.isAbsolute = path.win32.isAbsolute;
+        path.basename = path.win32.basename;
+        os.platform = _.constant('win32');
+      });
+
+      afterEach(function() {
+        path.isAbsolute = this.isAbsolute;
+        path.basename = this.basename;
+        os.platform = this.platform;
+      });
+
+      it('should replace absolute paths with the basename', function() {
+        const object = {
+          prop1: 'some value',
+          prop2: 'C:\\Users\\John\\rpi.img'
+        };
+
+        m.chai.expect(utils.hideObjectAbsolutePaths(object)).to.deep.equal({
+          prop1: 'some value',
+          prop2: 'rpi.img'
+        });
+
+      });
+
+      it('should replace nested absolute paths with the basename', function() {
+        const object = {
+          nested: {
+            path: 'C:\\Users\\John\\rpi.img'
+          }
+        };
+
+        m.chai.expect(utils.hideObjectAbsolutePaths(object)).to.deep.equal({
+          nested: {
+            path: 'rpi.img'
+          }
+        });
+      });
+
+      it('should not alter \\\\.\\PHYSICALDRIVE1', function() {
```

Comment:
Sorry, I meant maybe we should check that UNC paths also get **do** anonymized, and that the special-case code to not anonymize e.g. `\\.\PHYSICALDRIVE1` doesn't accidentally prevent e.g. `\\server\share\path\file` or `\\?\D:\very long path\file` from being anonymized?

> Or are they just a weird / rare case that we shouldn't care about?

Yeah, you might be right ;-)